eval_video_segmentation.py

```diff
@@ -107,37 +110,20 @@ def norm_mask(mask):
     return mask
 
 
-def label_propagation(args, model, frame_tar, list_frames, list_segs, mask_neighborhood=None):
+def label_propagation(args, model, frame_tar, list_frame_feats, list_segs, mask_neighborhood=None):
     """
     propagate segs of frames in list_frames to frame_tar
     """
-    # get feature maps
-    input_frames = torch.stack([frame_tar] + list_frames)
-    curr_i = 0
-    _, w0, h0 = frame_tar.shape
-    while curr_i < input_frames.shape[0]:
-        inp = input_frames[curr_i: curr_i + args.bs].cuda()
-        out = model.get_intermediate_layers(inp, n=1)[0]
-        out = out[:, 1:, :]  # we discard the [CLS] token
-        w, h = int(w0 / model.patch_embed.patch_size), int(h0 / model.patch_embed.patch_size)
-        dim = out.shape[-1]
-        out = out.reshape(out.shape[0], w, h, dim)
-        out = out.permute(0, 3, 1, 2)
-
-        if curr_i == 0:
-            feature_maps = out
-        else:
-            feature_maps = torch.cat((feature_maps, out))
-        curr_i += args.bs
-        del out  # free memory
+    ## we only need to extract feature of the target frame
+    feat_tar, h, w = extract_feature(model, frame_tar, return_h_w=True)
```

Comment:
I see two options to fix the bug:

1) `feat_tar, w, h = extract_feature(model, frame_tar, return_h_w=True)`

OR

2) Replace line 133 `mask_neighborhood = restrict_neighborhood(w, h)` by `mask_neighborhood = restrict_neighborhood(h, w)`

Reply:
I would recommend option 2.